seqr/views/utils/airflow_utils.py

```diff
@@ -63,16 +101,110 @@ def _check_dag_running_state(dag_id):
         raise DagRunningException(f'{dag_id} is running and cannot be triggered again.')
 
 
-def _construct_dag_variables(projects, data_path, additional_variables):
+def _construct_v2_dag_name(sample_type: str, dataset_type: str = Sample.DATASET_TYPE_VARIANT_CALLS,
+                           is_internal: bool = True, callset: str = 'Internal'):
+    if is_internal:
+        dag_dataset_type = f'_{_dag_dataset_type(sample_type, dataset_type)}'
+        return f'RDG_{sample_type}_Broad_{callset}{"" if dataset_type == Sample.DATASET_TYPE_VARIANT_CALLS else dag_dataset_type}'
+    return f'AnVIL_{sample_type}'
+
+
+def _construct_v3_dag_name(sample_type: str, dataset_type: str, **kwargs):
+    return f'v03_pipeline-{_dag_dataset_type(sample_type, dataset_type)}'
+
+
+def _construct_v2_dag_id(dag_name: str):
+    return f'seqr_vcf_to_es_{dag_name}_v{AIRFLOW_DAG_VERSION}'
+
+
+def _dag_dataset_type(sample_type: str, dataset_type: str):
+    return 'GCNV' if dataset_type == Sample.DATASET_TYPE_SV_CALLS and sample_type == Sample.SAMPLE_TYPE_WES \
+        else dataset_type
+
+
+def _construct_v2_dag_variables(projects: list[str], data_path: str, genome_version: str, is_internal: bool,
+                                dag_name: str, user: User, **kwargs):
     dag_variables = {
         "active_projects": projects,
         "projects_to_run": projects,
         "vcf_path": data_path,
     }
-    dag_variables.update(additional_variables)
+    dag_path = _get_dag_gs_path(genome_version, dag_name)
+    if is_internal:
+        version_paths = get_gs_file_list(dag_path, user=user, allow_missing=True, check_subfolders=False)
+        versions = [re.findall(f'{dag_path}/v(\d\d)/', p) for p in version_paths]
+        curr_version = max([int(v[0]) for v in versions if v] + [0])
+        dag_variables['version_path'] = f'{dag_path}/v{curr_version + 1:02d}'
+    else:
+        dag_variables['project_path'] = f'{dag_path}/{projects[0]}/v{datetime.now().strftime("%Y%m%d")}'
     return dag_variables
 
 
+def _construct_v3_dag_variables(projects: list[str], data_path: str, genome_version: str, is_internal: bool,
+                                sample_type: str, **kwargs):
+    return {
+        'projects_to_run': projects,
+        'callset_paths': [data_path],
+        'sample_source': 'Broad_Internal' if is_internal else 'AnVIL',
```

Comment:
I'm supporting `Broad_External` still in airflow.  It looks like there's a couple of exome projects in there that have subset files, but that's it.  Do we expect more?

Reply:
We should continue to support `Broad_External` in airflow, but its not currently supported being triggered from seqr, in v2 or v3. Currently we have the UI to support triggering either:
 - AnVIL SNV_INDEL DAGs
 - Broad_Internal SV/GCNV/MITO DAGs

Since we do not even have SV/GCNV/MITO for Broad_External, theres no reason to have it supported from the seqr UI/ backend. When we add support for triggering SNV_INDEL to the "internal" data triggering UI, we can think about if we want to support Broad_External from the UI or no.

Out of curiosity, how does the pieplein run differ between `Broad_Internal` and `Broad_External`?

Reply:
There's no difference beyond where airflow looks for sample subset, pedigree, and remap files.